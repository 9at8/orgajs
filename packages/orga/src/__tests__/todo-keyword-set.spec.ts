import todoKeywordSet from '../todo-keyword-set'

const debug = (text: string) => {
  const tks = todoKeywordSet(text)
  console.log(`${text}:`, tks)
}

describe('TodoKeywordSet', () => {
  it('works', () => {
    expect(todoKeywordSet('TODO | DONE')).toMatchSnapshot()
    expect(todoKeywordSet('TODO DONE')).toMatchSnapshot()
    expect(todoKeywordSet(' TODO NEXT  |  DONE ')).toMatchSnapshot()
    expect(todoKeywordSet('TODO NEXT DONE')).toMatchSnapshot()
    expect(todoKeywordSet('TODO | DONE CANCELLED')).toMatchSnapshot()
  })
})
