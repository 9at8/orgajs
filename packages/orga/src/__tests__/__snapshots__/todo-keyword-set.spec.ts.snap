// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoKeywordSet works 1`] = `
Object {
  "actionables": Array [
    "TODO",
  ],
  "done": Array [
    "DONE",
  ],
  "keywords": Array [
    "TODO",
    "DONE",
  ],
}
`;

exports[`TodoKeywordSet works 2`] = `
Object {
  "actionables": Array [
    "TODO",
  ],
  "done": Array [
    "DONE",
  ],
  "keywords": Array [
    "TODO",
    "DONE",
  ],
}
`;

exports[`TodoKeywordSet works 3`] = `
Object {
  "actionables": Array [
    "TODO",
    "NEXT",
  ],
  "done": Array [
    "DONE",
  ],
  "keywords": Array [
    "TODO",
    "NEXT",
    "DONE",
  ],
}
`;

exports[`TodoKeywordSet works 4`] = `
Object {
  "actionables": Array [
    "TODO",
    "NEXT",
  ],
  "done": Array [
    "DONE",
  ],
  "keywords": Array [
    "TODO",
    "NEXT",
    "DONE",
  ],
}
`;

exports[`TodoKeywordSet works 5`] = `
Object {
  "actionables": Array [
    "TODO",
  ],
  "done": Array [
    "DONE",
    "CANCELLED",
  ],
  "keywords": Array [
    "TODO",
    "DONE",
    "CANCELLED",
  ],
}
`;
